### PR TITLE
feat: add file upload scenario

### LINKED ISSUE #8: File upload
Scenario for uploading a file (doesn't need to do anything with it)

### PR DESCRIPTION
**Pull request by gregoryduckworth**
### Description
Adds a new scenario for file upload

### Related Issue
Closes #8 

### Checklist

- [x] I have read the contribution guidelines.
- [x] This PR includes tests (or updates existing tests) that cover the changes.
- [x] Documentation has been updated (if applicable).
- [x] Code follows the project's coding standards.
- [x] All tests pass.

### CODE REVIEW

File: src/scenarios/FileUpload.scenario.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        <Button
+          variant='contained'
+          color='primary'
+          onClick={handleUpload}
+          disabled={uploading}
+          data-testid='upload-button'
+        >
+          {t('scenarios.file-upload.upload-button')}
+        </Button>
+        {uploading && <LinearProgress sx={{ mt: 2 }} />}
```
Comment (e-taniguti) [inline src/scenarios/FileUpload.scenario.tsx:97]: How about showing the selected file ?
        <Typography variant='body2'>
            {file?.name}
        </Typography>

Comment (gregoryduckworth) [reply src/scenarios/FileUpload.scenario.tsx:97]: Added the filename to the translated string now.

image

Comment (e-taniguti) [reply src/scenarios/FileUpload.scenario.tsx:97]: Nice !!!
But I was thinking of showing the file name when it is selected. Before the upload.

Screenshot 2023-11-07 at 09 17 36

Comment (gregoryduckworth) [reply src/scenarios/FileUpload.scenario.tsx:97]: Ahh ok, will fix that now.

Comment (gregoryduckworth) [reply src/scenarios/FileUpload.scenario.tsx:97]: I've updated the behaviour now as well, so the upload button only appears when a file has been selected.

image

**Review by e-taniguti — APPROVED** [review]

**Pull request merged**